### PR TITLE
Allow to define the number of concurrent automatic replacements

### PR DESCRIPTION
**Pull request by johscheuer**
Fixes: https://github.com/FoundationDB/fdb-kubernetes-operator/issues/604 
Fixes: https://github.com/FoundationDB/fdb-kubernetes-operator/issues/695

### CODE REVIEW

File: controllers/replace_failed_pods.go [hunk 1]
```diff
@@ -52,29 +53,50 @@ func chooseNewRemovals(cluster *fdbtypes.FoundationDBCluster) bool {
...
 			// replacing more failed pods.
-			return false
+			removalCnt++
 		}
 	}
+	maxReplacements := cluster.GetMaxConcurrentReplacements() - removalCnt
 
+	hasReplacement := false
 	for _, processGroupStatus := range cluster.Status.ProcessGroups {
+		if maxReplacements == 0 {
```
Comment (johscheuer) [inline controllers/replace_failed_pods.go:71]:
```suggestion
		if maxReplacements <= 0 {
```

There are cases where we have a negative number --> Add a test case for that.

File: controllers/replace_failed_pods.go [hunk 2]
```diff
@@ -52,29 +53,50 @@ func chooseNewRemovals(cluster *fdbtypes.FoundationDBCluster) bool {
 		return false
 	}
 
+	// The maximum number of removals will be the defined number in the cluster spec
+	// minus all currently ongoing removals e.g. process groups marked fro removal but
+	// not fully excluded.
+	removalCnt := 0
```
Comment (brownleej) [inline controllers/replace_failed_pods.go:59]: Can we change this to `removalCount`? My brain consistently parses the word `cnt` as expanding to something else, so I would rather we spell it out 😊

**Review by brownleej — CHANGES_REQUESTED** [review]

File: controllers/replace_failed_pods.go (hunk 2 above)
Comment (johscheuer) [reply controllers/replace_failed_pods.go:59]: Sure :)

File: controllers/replace_failed_pods_test.go [hunk 3]
```diff
@@ -113,6 +113,36 @@ var _ = Describe("replace_failed_pods", func() {
...
+					It("should return true", func() {
+						Expect(result).To(BeTrue())
+					})
+
+					It("should mark the process group for removal", func() {
+						Expect(getRemovedProcessGroupIDs(cluster)).To(Equal([]string{"storage-2", "storage-3"}))
+					})
+				})
+
+				When("max concurrent replacements is set to one", func() {
```
Comment (johscheuer) [inline controllers/replace_failed_pods_test.go:132]:
```suggestion
				When("max concurrent replacements is set to zero", func() {
```

**Review by 09harsh — APPROVED** [review]
LGTM 👍

**Pull request merged**